ciphers/brainfuck.py

```diff
@@ -0,0 +1,195 @@
+        """
+        self.main_arr[self.data_ptr] += 1
+        self.main_arr[self.data_ptr] = self.main_arr[self.data_ptr] & 0xFF
+
+    def __decrement_data_value(self) -> None:
+        """
+        Decrement the byte at the data pointer by one.
+        """
+        self.main_arr[self.data_ptr] -= 1
+        self.main_arr[self.data_ptr] = self.main_arr[self.data_ptr] & 0xFF
+
+    def __append_bracket(self) -> None:
+        """
+        If the byte at the data pointer is zero, then instead of moving the
+        instruction pointer forward to the next command, jump it forward to
+        the command after the matching `]` command.
+        """
+        if self.main_arr[self.data_ptr] == 0:
+            self.instruction_ptr = self.loop_table[self.instruction_ptr]
+
+    def __pop_bracket(self) -> None:
+        """
+        If the byte at the data pointer is nonzero, then instead of moving the
+        instruction pointer forward to the next command, jump it back to the
+        command after the matching `[` command.
+        """
+        if self.main_arr[self.data_ptr] != 0:
+            self.instruction_ptr = self.loop_table[self.instruction_ptr]
+
+    def __print_output(self) -> None:
+        """
+        Output the byte at the data pointer.
+        """
+        self.output += chr(self.main_arr[self.data_ptr])
+
+    def __take_input(self) -> None:
+        """
+        Accept one byte of input, storing its value in the byte at the data
+        pointer.
+        """
+        if self.user_input == []:
+            user_input = list(input() + "\n")
+        self.main_arr[self.data_ptr] = ord(user_input.pop(0))
+
+    def bf_compiler(self) -> str:
+        """
+        Executes the brainfuck instructions and returns appropriate output after
+        compilation.
+
+        Returns:
+            `str`: A string representing the output of the BF instructions.
+        """
+        try:
+            self.__syntax_check()
+            while self.instruction_ptr < len(self.instruction):
+                if self.instruction[self.instruction_ptr] == ">":
+                    self.__increment_data_ptr()
+                elif self.instruction[self.instruction_ptr] == "<":
+                    self.__decrement_data_ptr()
+                elif self.instruction[self.instruction_ptr] == "+":
+                    self.__increment_data_value()
+                elif self.instruction[self.instruction_ptr] == "-":
+                    self.__decrement_data_value()
+                elif self.instruction[self.instruction_ptr] == "[":
+                    self.__append_bracket()
+                elif self.instruction[self.instruction_ptr] == "]":
+                    self.__pop_bracket()
+                elif self.instruction[self.instruction_ptr] == ".":
+                    self.__print_output()
+                elif self.instruction[self.instruction_ptr] == ",":
+                    self.__take_input()
```

Comment:
This needs timeit benchmarks because it would be a big performance gain to perform dict lookup of instructions.
```suggestion
instructions = {
    "-": self.__decrement_data_value,
    ",": self.__take_input,
    ".": self.__print_output,
    "[": self.__append_bracket,
    "]": self.__pop_bracket,
    "+": self.__increment_data_value,
    "<": self.__decrement_data_ptr,
    ">": self.__increment_data_ptr,
}
if method := instructions.get(self.instruction[self.instruction_ptr])
    method()
else:
    msg = f"{self.instruction[self.instruction_ptr]} is an invalid instruction"
    raise ValueError(msg)
```